### PR TITLE
[#151886563] use simple query string query for full-text searches

### PR DESCRIPTION
**Pull request by MothOnMars**
- bump some gems
- add debug logging for ES queries
- refactor tag search specs

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -19,14 +19,14 @@ gem 'elasticsearch-model', '~>0.1.0'
 gem 'elasticsearch-dsl', '~> 0.1.5'
 
 gem 'newrelic_rpm', '~> 4.2'
-gem 'airbrake', '~> 6.1'
+gem 'airbrake', '~> 7.1'
```
Comment (MothOnMars) [inline Gemfile]: @noremmie - heads up on this Airbrake upgrade. I don't see any potentially breaking changes, but you never know: https://github.com/airbrake/airbrake/blob/master/CHANGELOG.md

Comment (noremmie) [reply Gemfile]: Thanks, @MothOnMars. After the next i14y deploy perhaps we can tickle a known error to make sure it shows up in Airbrake.

**Comment by jamesisaacs** [comment]
@MothOnMars LGTM! After we upgrade ES, we'll have to remove the now deprecated `filtered query` and use the `filter` parameter within the `bool query`. I'm also curious about whether we want score explanations and the current cutoff frequencies but that can wait until later.

**Comment by MothOnMars** [comment]
Yeah, I figure there will be plenty more changes to come. Once we are on ES5 and also have a sense of where we want to tweak the search scoring, we can do some brainstorming. (Note: score explanations are already enabled in debug mode: https://github.com/GSA/i14y/blob/287696099f02085a39ffb158866d418836d2af24/app/classes/document_query.rb#L34)

**Pull request merged**